packages/nimble-components/src/table-column/duration-text/models/duration-formatter.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * A class for formatting a numeric value in the unit of seconds into a display value representing its
+ *  days, hours, minutes, and seconds.
+ */
+export class DurationFormatter {
+    private readonly daysFormatter: Intl.NumberFormat;
+    private readonly hoursFormatter: Intl.NumberFormat;
+    private readonly minutesFormatter: Intl.NumberFormat;
+    private readonly secondsFormatter: Intl.NumberFormat;
+    private readonly scientificSecondsFormatter: Intl.NumberFormat;
+
+    public constructor(private readonly lang: string) {
+        this.daysFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'day'
+        });
+        this.hoursFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'hour'
+        });
+        this.minutesFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'minute'
+        });
+        this.secondsFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'second'
+        });
+        this.scientificSecondsFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'second',
+            notation: 'scientific',
+            maximumFractionDigits: 3
+        });
+    }
+
+    public format(value: number | null | undefined): string {
+        if (
+            value === null
+            || value === undefined
+            || value < 0
+            || !Number.isFinite(value)
+        ) {
+            return '';
+        }
+
+        const result = [];
+        const fractionalDays = value / 86400000;
+        let remainingTime = value;
+        const days = Math.floor(fractionalDays);
+        if (days <= 100 && days > 0) {
+            const formattedDays = this.daysFormatter.format(days);
+            result.push(formattedDays);
+            remainingTime -= 86400000;
```

Comment:
You are right, however I couldn't think of a new test that would have caught the incorrect previous algorithm. Ultimately, I think the previous algorithm worked because of how it is ultimately always taking a modded value to work from, so it didn't matter what multiple of each "time segment" was used to subtract from the remaining time.